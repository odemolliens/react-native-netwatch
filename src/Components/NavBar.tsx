import { Appbar } from 'react-native-paper';
import { Platform, StyleSheet, TouchableOpacity } from "react-native";
import FeatherIcon from 'react-native-vector-icons/Feather';
import * as React from 'react';
import { ReactNode, useContext } from 'react';
import { ThemeContext } from '../Theme';

export function NavBar(props: {
  title: string;
  rightComponent?: ReactNode;
  icon?: string;
  onPressBack: (showDetails: boolean) => void;
}) {
  const theme = useContext(ThemeContext);

  return (
    <Appbar.Header style={[styles.header, { backgroundColor: theme.secondaryDarkColor }]}>
      <TouchableOpacity
        style={[styles.button, { borderLeftWidth: 0 }]}
        onPress={() => props.onPressBack(false)}
        testID="buttonBackToMainScreen"
      >
        <FeatherIcon name={props.icon || 'arrow-left'} color={theme.textColorOne} size={24} />
      </TouchableOpacity>
      <Appbar.Content color={theme.primaryColor} title={props.title} titleStyle={{ fontSize: 18 }} />
      {props.rightComponent}
    </Appbar.Header>
  );
}

const styles = StyleSheet.create({
  header: {
    paddingTop: Platform.OS === 'ios' ? 0 : 40,
    elevation: 0,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
    height: Platform.OS === 'ios' ? 60 : 100,
  },

  container: {
    position: 'absolute',
    height: '100%',
    width: '100%',
    top: 0,
    left: 0,
  },

  button: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 56,
    width: 56,
    borderLeftWidth: 1,
  },
});
