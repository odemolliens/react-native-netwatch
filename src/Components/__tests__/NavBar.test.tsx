import React from 'react';
import { NavBar } from '../NavBar';
import { shallow, ShallowWrapper } from 'enzyme';
import { TouchableOpacity } from 'react-native';

describe('Test NabBar componsnet', () => {
  let component: ShallowWrapper;

  it('should render properly', () => {
    component = shallow(<NavBar title={'title'} icon={'left'} onPressBack={() => jest.fn()} />);
    expect(component).toMatchSnapshot();
  });

  it('should go back when pressing left side', () => {
    const onPressBack = jest.fn();
    component = shallow(<NavBar title={'title'} icon={'left'} onPressBack={onPressBack} />);
    component.find(TouchableOpacity).simulate('press');
    expect(onPressBack).toHaveBeenCalled();
  });
});
