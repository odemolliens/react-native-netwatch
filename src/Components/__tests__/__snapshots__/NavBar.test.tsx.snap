// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NabBar componsnet should render properly 1`] = `
<Header
  style={
    Array [
      Object {
        "elevation": 0,
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
      },
      Object {
        "backgroundColor": "#111827",
      },
    ]
  }
>
  <ForwardRef
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderLeftWidth": 1,
          "height": 56,
          "justifyContent": "center",
          "width": 56,
        },
        Object {
          "borderLeftWidth": 0,
        },
      ]
    }
    testID="buttonBackToMainScreen"
  >
    <Icon
      color="#F9FAFB"
      name="left"
      size={24}
    />
  </ForwardRef>
  <Content
    color="#06B6D4"
    title="title"
    titleStyle={
      Object {
        "fontSize": 18,
      }
    }
  />
</Header>
`;
